### PR TITLE
fix #172

### LINKED ISSUE #172: triggered events with more than one argument
If an event is ever emitted that accepts more than one arg, the angular component will only emit the first:

https://github.com/fullcalendar/fullcalendar-angular/blob/bff9072ee732588666a74c2ea082987967617e99/projects/fullcalendar/src/lib/fullcalendar.component.ts#L49

Does angular allow emitting more than one arg? If so, can we change that line to emit ALL args, like this:

https://github.com/fullcalendar/fullcalendar-vue/blob/e8b7e1a6182061aacc055f64844498beb1be4fd7/src/FullCalendar.js#L49

### PR DESCRIPTION
**Pull request by irustm**

### CODE REVIEW

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,85 @@
...
+  ></full-calendar>`
+})
+class TestHostComponent {
+  plugins = [dayGridPlugin];
+  header: {
+    left: 'prev,next today myCustomButton',
+    center: 'title',
+    right: 'dayGridMonth'
+  };
+  weekends = true;
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is `weekendsEnabled`

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,85 @@
...
+  viewSkeletonRender = false;
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is `viewSkeletonRendered`

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,85 @@
...
+  updateWeekendsView() {
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is `disableWeekends()`

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 4]
```diff
@@ -1,25 +1,85 @@
...
+    this.weekends = false;
+  }
+  viewSkeletonRenderEvent(event) {
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is `handleViewSkeletonRender`

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 5]
```diff
@@ -1,25 +1,85 @@
...
+@Component({
+  selector: `full-calendar-test`,
+  template: `<full-calendar
+  [plugins]="plugins"
+  [header]="header"
+  [weekends]="weekends"
+  (viewSkeletonRender)="viewSkeletonRenderEvent($event)"
+  ></full-calendar>`
+})
+class TestHostComponent {
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is just `HostComponent`. "Test" is unnecessary

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 6]
```diff
@@ -1,25 +1,85 @@
...
+    this.viewSkeletonRender = true;
+  }
+}
 
 describe('FullCalendarComponent', () => {
   let component: FullCalendarComponent;
   let fixture: ComponentFixture<FullCalendarComponent>;
 
+  let testComponent: TestHostComponent;
+  let testFixture: ComponentFixture<TestHostComponent>;
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: a better name is `hostComponent` and `hostFixture`

File: projects/fullcalendar/src/lib/fullcalendar.component.spec.ts [hunk 7]
```diff
@@ -1,25 +1,85 @@
...
+    testFixture = TestBed.createComponent(TestHostComponent);
+    testComponent = testFixture.componentInstance;
+    testFixture.detectChanges();
+  });
 
   it('should create', () => {
     expect(component).toBeTruthy();
+    expect(fixture.nativeElement.querySelector('.fc-toolbar')).toBeTruthy();
+  });
+  it('should unmount and call destroy', () => {
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: please add blank lines in between the `it` statements

Comment (arshaw) [reply projects/fullcalendar/src/lib/fullcalendar.component.spec.ts]: the test "should unmount and call destroy" does NOT actually test that "destroy" was called. It SHOULD. It only checks that the fc-toolbar is gone.

Make a handler for the `_destroy` event in HostFixture that sets a property called `wasDestroyed` to `true`. Then, for the test, check that `wasDestroyed` is `true`.

File: projects/fullcalendar/src/lib/fullcalendar.component.ts [hunk 8]
```diff
@@ -45,8 +45,8 @@ export class FullCalendarComponent implements AfterViewInit, OnChanges, OnDestro
     const options = {};
 
     EVENT_NAMES.forEach(eventName => {
-      options[eventName] = info => {
-        this[eventName].emit(info); // TODO: handle more than one arg?
+      options[eventName] = (...args) => {
+        this[eventName].emit(...args); // TODO: handle more than one arg?
```
Comment (arshaw) [inline projects/fullcalendar/src/lib/fullcalendar.component.ts]: remove the comment `// TODO: handle more than one arg?`

**Comment by irustm** [comment]
@arshaw 
Thank you for watching
I'm sorry I didn't wait for your comments,
I'm gonna do it.

**Pull request merged**